### PR TITLE
[17.0][MIG] mail_composer_cc_bcc: Migration to 17.0

### PR DESCRIPTION
**Pull request by trisdoan**
### 1) Summary of changes in 17.0

mail.template
https://github.com/odoo/odoo/blob/17.0/addons/mail/models/mail_template.py

- `generate_recipients` replaced by `_generate_template_recipients`

mail.mail:
https://github.com/odoo/odoo/blob/17.0/addons/mail/models/mail_mail.py

- new method `_prepare_outgoing_list`: https://github.com/odoo/odoo/commit/b1d678b12ac5aac4086771f9ae88d401d919c9fa
- replaces `_send_prepare_values`
- new method `_personalize_outgoing_body` is also introduced

account.move.send: replaces account_invoice_send
- do not inherit mail.compose.message
- `_action_send_mail` replaced by `_send_mail`
- introduce new method `_get_mail_move_values`

### 2) Consequences
- We don't need to fully replace mail.mail:_send_mail() as we did before, thanks to _prepare_outgoing_list.
- override `_get_mail_move_values` and `_send_mail` in account.move.send as it does not benefit from mail.compose.message

### 3) This change

- roll back to do as odoo: one mail per recipient
- keep same all headers (To, Cc, Bcc) in all emails
- use a context to define the proper smtp_to for each email

### CODE REVIEW

File: mail_composer_cc_bcc/models/mail_thread.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models, tools
+
+
+def format_emails(partners):
+    emails = [
+        tools.formataddr((p.name or "False", p.email or "False")) for p in partners
+    ]
+    return ", ".join(emails)
```
Comment (nilshamerlinck) [inline mail_composer_cc_bcc/models/mail_thread.py]: weird, can you check when did we introduce that duplicate?

should be `from .mail_mail import format_emails`

Comment (trisdoan) [reply mail_composer_cc_bcc/models/mail_thread.py]: Probably I duplicated it when migrating. I just changed to `from .mail_mail import format_emails`

**Comment by nilshamerlinck** [comment]
- nitpick: small typo in commit name: `[IMP] mail_composer_cc_bcc: pre-comit`
- created a PR on this branch to fine tune things a bit for the Bcc + add some debug logging for the rcpt_to
- then I think we're almost done ;-)
- ok for you to define yourself as `maintainer` of that module?

**Comment by trisdoan** [comment]
Hi anh, thank you so much for your support
yes, maintainer ok for me

File: mail_composer_cc_bcc/__manifest__.py [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
     "name": "Email CC and BCC",
     "summary": "This module enables sending mail to CC and BCC partners in mail composer form.",
-    "version": "16.0.1.0.1",
+    "version": "17.0.1.0.0",
     "development_status": "Alpha",
     "category": "Social",
     "website": "https://github.com/OCA/social",
     "author": "Camptocamp SA, Odoo Community Association (OCA)",
-    "maintainers": ["hailangvn2023"],
+    "maintainers": ["hailangvn2023", "trisdoan"],
```
Comment (nilshamerlinck) [inline mail_composer_cc_bcc/__manifest__.py]: you can remove Hai from here

**Review by nilshamerlinck — APPROVED** [review]

File: mail_composer_cc_bcc/__manifest__.py (hunk 2 above)
Comment (nilshamerlinck) [reply mail_composer_cc_bcc/__manifest__.py]: in the final PR description, you can add something like: "My colleague Hai is busy on other topics, I will take over as a maintainer of this module from now on."

**Pull request closed without merging**